packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -83,12 +83,15 @@ function classOrObjectMethod(path: NodePath, callId: Object) {
 
   node.async = false;
 
-  const container = t.functionExpression(null, [], t.blockStatement(body.body), true);
+  const container = t.functionExpression(node.key, [], t.blockStatement(body.body), true);
   container.shadow = true;
   body.body = [
     t.returnStatement(t.callExpression(
-      t.callExpression(callId, [container]),
```

Comment:
@loganfsmyth I did some testing in Chrome, Safari, and Firefox using a [test case I made on JSPerf](https://jsperf.com/calling-function-with-no-args), and it looks like `.call` might be the best option for no arguments based on those tests (though I'm not sure how much real-world scenarios will differ from my simple test case).